src/include/migraphx/matcher.hpp

```diff
@@ -896,10 +897,20 @@ inline auto has_value(T x, std::size_t atol_mult = 10, std::size_t rtol_mult = 1
             else
             {
                 auto eps = std::numeric_limits<type>::epsilon();
-                if(std::all_of(v.begin(), v.end(), [&](auto val) {
-                       return std::fabs(val - static_cast<type>(x)) <
-                              eps * (atol_mult + rtol_mult * std::fabs(val));
-                   }))
+                auto tolerance = atol_mult + rtol_mult * std::fabs(x);
+                if(migraphx::float_equal(tolerance, 0.0))
+                {
+                    // do exact match using float_equal if tolerance is zero
+                    if(std::all_of(v.begin(), v.end(), [&](auto val) {
+                           return migraphx::float_equal(val, static_cast<type>(x));
```

Comment:
`migraphx::float_equal` is also an incorrect implementation. That should be fixed (either in this PR, or separately..) 

float_equal_fn::apply() should be rewritten.
It is:    
```
static bool apply(T x, T y)
    {
        return std::isfinite(x) and std::isfinite(y) and
               std::nextafter(x, std::numeric_limits<T>::lowest()) <= y and
               std::nextafter(x, std::numeric_limits<T>::max()) >= y;
    }
```
And it should be:
```
   static bool apply(T x, T y)
    {
        return std::isfinite(x) and std::isfinite(y) and
               y >= std::nextafter(x, std::numeric_limits<T>::lowest()) and
               y <= std::nextafter(x, std::numeric_limits<T>::max());
    }
```
Thanks.